src/tools.py

```diff
@@ -64,29 +61,21 @@ def tools_rootpw(new_password, check_strength=True):
     if check_strength:
         assert_password_is_strong_enough("admin", new_password)
 
-    new_hash = _hash_user_password(new_password)
-
-    # Write as root password
-    try:
-        hash_root = spwd.getspnam("root").sp_pwd
-
-        with open("/etc/shadow", "r") as before_file:
-            before = before_file.read()
+    proc = subprocess.Popen(
+        ["passwd"],
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+    )
+    out, err = proc.communicate(f"{new_password}\n{new_password}\n".encode("utf-8"))
+    result = proc.wait()
```

Comment:
These days apparently it's more practical to use `subprocess.run()` I think ? Maybe something like : 

```
out, err = subprocess.run(["passwd"], input=f"{new_password}\n{new_password}\n".encode("utf-8"), capture_output=True)
```

Reply:
Ah didn't know you could pass `input` like that ! yeah feels about right.